libc/src/__support/CPP/new.h

```diff
@@ -14,7 +14,7 @@
 #include "src/__support/macros/properties/os.h"
 
 #include <stddef.h> // For size_t
-#include <stdlib.h> // For malloc, free etc.
```

Comment:
So what do we still need this for then?

Reply:
I think we will need some declaration for malloc/free in their own proxy headers.

Reply:
Ok, but in that case, the comment should not be deleted in this PR.

Reply:
i will keep the comment, and i will do the proxy for malloc and free in this pr

Reply:
You can do that in a new PR and land that first, then revisit this one. That will keep this PR more focused perhaps.

Reply:
ok, i will get the proxy for malloc and free done before this pr. thank you

Reply:
I’m going to have to do this pr first because the malloc and free proxies depend on the stdlib_overlay.h file that’s in this pr. I have to add it to the CMakelists in hdr/. Is this ok with you? Or do you have suggestions?

Reply:
ah, ok. No that's fine then. Thanks for considering the suggestion.